processor/src/chiplets/ace/trace.rs

```diff
@@ -47,13 +51,15 @@ impl CircuitEvaluation {
     /// Generates the nodes in the graph generated by evaluating the inputs and circuit
     /// located in a contiguous memory region.
     pub fn new(ctx: ContextId, clk: RowIndex, num_read_rows: u32, num_eval_rows: u32) -> Self {
-        let num_wires = 2 * num_read_rows + num_eval_rows;
+        let num_wires = 2 * (num_read_rows as u64) + (num_eval_rows as u64);
+        assert!(num_wires <= MAX_ACE_WIRES as u64, "too many wires");
+
         let num_rows = num_read_rows + num_eval_rows;
 
         Self {
             ctx,
             clk,
-            wire_bus: WireBus::new(num_wires),
+            wire_bus: WireBus::new(num_wires as u32),
```

Comment:
Question: would it also make sense to check that it is safely castable as a u32?

Reply:
The maximum is 2^30 so casting is always safe.